indico/modules/events/editing/client/js/management/editable_type/TeamManagerModal.jsx

```diff
@@ -0,0 +1,93 @@
+
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Form as FinalForm} from 'react-final-form';
+import {Button, Form, Loader, Message, Modal} from 'semantic-ui-react';
+import {Translate} from 'indico/react/i18n';
+import {FinalPrincipalList} from 'indico/react/components';
+import {getChangedValues, FinalSubmitButton, FinalUnloadPrompt} from 'indico/react/forms';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {useFavoriteUsers, useIndicoAxios} from 'indico/react/hooks';
+import {useNumericParam} from 'indico/react/util/routing';
+
+export default function TeamManager({editableType, onClose}) {
+  const eventId = useNumericParam('confId');
+
+  const {data: principals, loading: isLoadingPrincipals} = useIndicoAxios({
+    url: principalsURL({confId: eventId, type: editableType}),
+    trigger: eventId,
+  });
+
+  const handleSubmit = async (data, form) => {
+    const changedValues = getChangedValues(data, form);
+    try {
+      await indicoAxios.post(principalsURL({confId: eventId, type: editableType}), changedValues);
+    } catch (error) {
+      handleAxiosError(error);
+    }
+    onClose();
+  };
+  const favoriteUsersController = useFavoriteUsers();
+
+  if (isLoadingPrincipals) {
+    return <Loader active />;
+  } else if (!principals) {
+    return null;
+  }
+
+  return (
+    <div style={{padding: '30px'}}>
+      <FinalForm
+        onSubmit={handleSubmit}
+        initialValues={principals}
+        initialValuesEqual={_.isEqual}
+        subscription={{}}
+      >
+        {fprops => (
+          <Modal onClose={onClose} size="small" closeOnDimmerClick={false} open>
+            <Modal.Header content={Translate.string('Set editing managers')} />
+            <Modal.Content>
+              <Form id="team-manager-form" onSubmit={fprops.handleSubmit}>
+                <FinalUnloadPrompt />
+                <Message>
+                  <FinalPrincipalList
+                    name="principals"
+                    withGroups
+                    withEventRoles
+                    withCategoryRoles
+                    favoriteUsersController={favoriteUsersController}
+                    label={Translate.string('Editing managers')}
+                    eventId={eventId}
+                  />
+                </Message>
+              </Form>
+            </Modal.Content>
+            <Modal.Actions style={{display: 'flex', justifyContent: 'flex-end'}}>
+              <FinalSubmitButton form="team-manager-form" label={Translate.string('Submit')} />
+              <Button onClick={onClose} disabled={fprops.submitting}>
+                <Translate>Cancel</Translate>
+              </Button>
+            </Modal.Actions>
+          </Modal>
+        )}
+      </FinalForm>
+    </div>
+  );
+}
+
+TeamManager.propTypes = {
+  editableType: PropTypes.string.isRequired,
+  onClose: PropTypes.func,
```

Comment:
I wouldn't set this as null in the default props, since we end up calling the function without checking it. IMO, I would suggest setting it as required. As we do it in other similar modals.

```suggestion
  onClose: PropTypes.func.isRequired,
```